save-backend/src/main/kotlin/org/cqfn/save/backend/security/ProjectPermissionEvaluator.kt

```diff
@@ -27,24 +35,55 @@ class ProjectPermissionEvaluator {
             Permission.DELETE -> false
         }
 
-        if (authentication.hasRole(Role.ADMIN)) {
+        if (authentication.hasRole(Role.SUPER_ADMIN)) {
             return true
         }
 
         val userId = (authentication.details as AuthenticationDetails).id
         return when (permission) {
             Permission.READ -> project.public || hasWriteAccess(userId, project)
             Permission.WRITE -> hasWriteAccess(userId, project)
+            // fixme: check role to ensure its `OWNER`
             Permission.DELETE -> project.userId == userId
```

Comment:
I mean, we should somehow check data for consistency: role `OWNER` is specifically for project owner, but we also have `project.user_id` for project owner. We should resolve this redundancy by removing one of the entities or by adding checks.